src/HeckeTypes.jl

```diff
@@ -359,6 +361,7 @@ mutable struct SRow{T, S} # S <: AbstractVector{T}
 
   function SRow{T, S}(A::SRow{T, S}; copy::Bool = false) where {T, S}
     copy || return A
+    @assert Vector{T} == S
```

Comment:
Another paranoia check that was valuable. It should get optimized away, I think?

Of course we could also try to make this function work generically, by doing something like this:
```julia
  return new{T, S}(base_ring(A), S(A.values), copy(A.pos))
```
(and adding a suitable copy constructor `ZZRingElem_Array(A:: ZZRingElem_Array)`

Reply:
Indeed, using `===` is a good idea here.
```suggestion
    @assert Vector{T} === S
```